### ISSUE TITLE
ImGUI light direction setting issue

### ISSUE THREAD

**Issue by DQLin** [NVIDIAGameWorks/Falcor#240]
In GuiImpl::addDirectionWidget, the variable for light direction is quantized to a step size of 0.001, but it is always normalized after that. This creates a problem: the value of light direction can change back and forth between the quantized value and normalized value with some user input. One consequence of this is that Scene::UpdateFlags::LightsMoved is always set to true when user change light direciton value through UI to some specific values (g.direction("Direction", mData.dirW) in DirectionalLight::renderUI, Light.cpp).

**Comment by kyaoNV** [comment]
This was recently fixed internally and will be included in the next release. The function now looks like:
```c++
    bool GuiImpl::addDirectionWidget(const char label[], float3& direction)
    {
        float3 dir = glm::normalize(direction);
        bool b = addVecVar(label, dir, -1.f, 1.f, 0.001f, false, "%.3f");
        if (b) direction = glm::normalize(dir);
        return b;
    }
```

**Comment by skallweitNV** [comment]
Fixed with 4.4.
